iterreg/sparse/solvers.py

```diff
@@ -194,16 +194,17 @@ def cd(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000,
             all_w[t // f_store] = w
             if verbose:
                 print(t, E[t // f_store])
-            if ES and (t > 0):
-                if np.abs(E[t // f_store - 1]-E[t // f_store]) < 1e-10 * E[0]:
-                    E[(t // f_store):] = E[t // f_store]
-                    all_w[(t // f_store):] = w
-                    return w, all_w, E
+
+            if (t > 0) and (np.abs(E[t // f_store - 1]-E[t // f_store]) <
+                            np.maximum(tol, 1e-10) * E[0]):
```

Comment:
why not just tol ? 
(also use the built in max function for float comparison if speed is not an issue, np.maximum is pointwise between two arrays)

Reply:
I thought that's what you said with "add a sclar tol parameter, equal to 0 by default, and put tol instead if 1e-10 below"

Reply:
Maybe it was "of"

Reply:
yes my bad indeed